### PR TITLE
chore: Define changelog entries for PRs merged after 1.15.3 and note for new changelog format

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-241733

Following new changelog guidelines defined in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2123 this PR includes changelog entries for PRs merged after our last release (1.15.3).

Note:
- Updating entries for multiple merged PRs is an exception, each PR will handle its changelog entries moving forward.
- Updating the CHANGELOG.md is also an exception for this PR needed to set the new Unreleased header. When CLOUDP-241727 is completed PRs will only include the changelog entry files (under `/.changelog`) and then an automated commit will update the changelog file.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .changelog/2100.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
```
Comment (lantoli) [inline .changelog/2100.txt:1]: it's ok although i thought you'll use a PR per file

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,22 @@
-# Changelog
+## (Unreleased)
```
Comment (lantoli) [inline CHANGELOG.md:1]: I understand this will be updated automatically in master once this project is in place

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+
+NOTES:
+
+* provider: New changelog format has been incorporated following [Terraform Changelog Specification](https://developer.hashicorp.com/terraform/plugin/best-practices/versioning#changelog-specification) ([#2124](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2124))
+
+ENHANCEMENTS:
+
+* data-source/mongodbatlas_backup_compliance_policy: Adds `policy_item_yearly` attribute ([#2109](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2109))
```
Comment (lantoli) [inline CHANGELOG.md:9]: i like how clean the changelog is

**Review by lantoli — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (AgustinBettati) [reply CHANGELOG.md:1]: correct, left a note in the PR description

File: .changelog/2100.txt (hunk 1 above)
Comment (AgustinBettati) [reply .changelog/2100.txt:1]: noted in PR description, this is an exception as I am covering several PRs to be "up to date", moving forward each PR will handle its changelog entry if needed.

**Review by oarbusi — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

**Review by EspenAlbert — APPROVED** [review]

**Review by maastha — APPROVED** [review]

**Review by corryroot — APPROVED** [review]
LGTM!

**Pull request merged**